core/app/modules/kit-library/data/repository.php

```
@@ -33,37 +38,54 @@ class Repository {
 	 * @throws Wp_Error_Exception
 	 */
 	public function get_all( $force_api_request = false ) {
+		$favorites = $this->user_favorites->get( 'elementor', 'kits' );
+
 		return $this->get_kits_data( $force_api_request )
-			->map( function ( $kit ) {
-				return $this->transform_kit_api_response( $kit );
+			->map( function ( $kit ) use ( $favorites ) {
+				return $this->transform_kit_api_response( $kit, $favorites );
 			} );
 	}
 
 	/**
 	 * Get specific kit.
 	 *
-	 * @param $id
+	 * @param       $id
+	 * @param array $options
 	 *
-	 * @return array|null
+	 * @return array
 	 * @throws Wp_Error_Exception
 	 */
-	public function find( $id ) {
+	public function find( $id, $options = [] ) {
+		$options = wp_parse_args( $options, [
+			'manifest_included' => true,
+		] );
+
 		$item = $this->get_kits_data()
 			->find( function ( $kit ) use ( $id ) {
 				return $kit->_id === $id;
 			} );
 
 		if ( ! $item ) {
-			return null;
+			throw new Wp_Error_Exception(
```

A `find` method should throw an error?